packages/commonwealth/client/scripts/hooks/api/queryKeys.ts

```diff
@@ -0,0 +1,6 @@
+const queryKeys = {
+  BULK_OFFCHAIN: (chainId) => `bulk-offchain-${chainId}`,
+  TOPICS_FOR_CHAIN: (chainId) => `${chainId}-topics`,
+};
+
```

Comment:
keys would define how cache is handled.

A key for dydx, would look like this `bulk-offchain-dydx`, and `bulk-offchain-ethereum` would be for ethereum.

RQ uses these keys for queries. So for example 
1. we visit `/dydx`
2. a `/bulkOffChain` query will be triggered and data will be cached in `bulk-offchain-dydx`
3. new we visit `ethereum`, a `/bulkOffChain` query will be triggered and data will be cached in `bulk-offchain-ethereum`. Now we have cache data for 2 communities.
4. If we visit `/dydx` back again, cached data will be used without making a new api call. A new call will be automatically triggered after the cache time expires

Reply:
This is super cool!

Reply:
More specific keys are definitely good. +1, lets go for it.

Reply:
Maybe we could also add a wrapper around the http lib (axios, or the jquery one) as well, that would hook up common auth tokens and headers config with each request.

We could also export the apis urls as some defined methods (on the api resource), something like
```jsx
const Topic = {
   create: (....args) => customHttpLib.post(`/some url`, ...args),
   update: (....args) => customHttpLib.post(`/some url`, ...args),
   delete: (....args) => customHttpLib.post(`/some url`, ...args),
   getById: (....args) => customHttpLib.post(`/some url`, ...args),
   getAll: (....args) => customHttpLib.post(`/some url`, ...args),
}
```
and use it like
```jsx
    useQuery({
      queryKey: [ApiRoutes.TopicById, { chainId, topicId }],
      queryFn: async () => {
        const { data } = await Topic.getById(topicId); // diff
        return data;
      },
    });
```
All endpoint would have these methods, so it be easy to follow along instead of just hardcoding the urls in each file directly